climada/hazard/relative_cropyield.py

```diff
@@ -0,0 +1,710 @@
+            for ax0, col in zip(axes, cols):
+                ax0.set_title(col, size='large')
+
+        return axes
+
+
+def generate_full_hazard_set(input_dir=INPUT_DIR, output_dir=OUTPUT_DIR, bbox=BBOX,
+                             isimip_run='ISIMIP2b', yearrange_his=None, yearrange_mean=None,
+                             return_data=False, save=True):
+
+    """Wrapper to generate full hazard set and save it to output directory.
+
+        Optional Parameters:
+            input_dir (string): path to input data directory
+            output_dir (string): path to output data directory
+            bbox (list of four floats): bounding box:
+                [lon min, lat min, lon max, lat max]
+            isimip_run (string): name of the ISIMIP run (ISIMIP2a or ISIMIP2b)
+            yearrange_his (int tuple): year range for the historical hazard sets
+            yearrange_mean (int tuple): year range for the historical mean
+            return_data (boolean): returned output
+                False: returns list of filenames only
+                True: returns also list of data
+            save (boolean): save output data to output_dir
+
+        Return:
+            filename_list (list): list of filenames
+
+        Optional Return:
+            output_list (list): list of generated output data (hazards and historical mean)
+
+    """
+
+    filenames = [f for f in listdir(input_dir) if (isfile(join(input_dir, f))) if not
+                 f.startswith('.')]
+
+
+    # generate output directories if they do not exist yet
+    if not os.path.exists(output_dir):
+        os.mkdir(output_dir)
+    if not os.path.exists(os.path.join(output_dir, 'Hazard')):
+        os.mkdir(os.path.join(output_dir, 'Hazard'))
+    if not os.path.exists(os.path.join(output_dir, 'Hist_mean')):
+        os.mkdir(os.path.join(output_dir, 'Hist_mean'))
+
+    filename_list = list()
+    output_list = list()
+
+    (his_file_list, file_props, hist_mean_per_crop,
+     scenario_list, crop_list) = init_hazard_set(filenames, input_dir, bbox, isimip_run,
+                                                 yearrange_his)
+
+    if yearrange_mean is None:
+        yearrange_mean = (YEARCHUNKS[(file_props[his_file_list[0]])['scenario']])['yearrange_mean']
+
+    for his_file in his_file_list:
+        haz_his, filename, hist_mean = calc_his_haz(his_file, file_props, input_dir, bbox,
+                                                    yearrange_mean)
+        # save the historic mean depending on the crop-irrigation combination
+        # the idx keeps track of the row in which the hist_mean values are written per crop-irr to
+        # ensure that all files are assigned to the corresponding crop-irr combination
+        hist_mean_per_crop[(file_props[his_file])['crop_irr']]['value'][
+            hist_mean_per_crop[(file_props[his_file])['crop_irr']]['idx'], :] = hist_mean
+        hist_mean_per_crop[file_props[his_file]['crop_irr']]['idx'] += 1
```

Comment:
Why add one to the mean?

Reply:
It is not adding 1 to the mean but rather to a counting index 'idx' within that loop. this rather complex way forward ensures flexibilit for users of that method to put any combination of ISIMIP output files (netcdf) in a folder and let CLIMADA create all hazard sets from that. Users scared of outsourcing this complexity handling can create the hazard files one by one from the single netcdf files.

Reply:
Then you should use a loop with enumerate() to make it clear that you are just counting the elements in the loop, right?

Reply:
The problem is, that the loop is over the files in the given input directory and not over the different crop-irrigation combinations (and the mean is supposed to be saved per crop-irrigation combination). Because of the filename structure the crop-irr of the files actually alternate constantly (and for some model-combinations they only exist for certain crops). So this was a way to ensure that the historical mean is matched to the according crop-irr combination. I tried to explain it in the comment above those lines of code, but it's a bit confusing.